Append bash-specific instructions to gigasecond (#611)
exercism/bash: exercises/practice/gigasecond/.docs/instructions.append.md

# placeholder

## How to solve with bash

`bash` does not have any builtin ways to **parse** datetime strings.
This exercise requires you to call out to an external utility.

* The GNU `date` command's `-d` option can parse [a wide variety of formats][gnu-date].
* For Mac users, see the [BSD `date` command][bsd-date] and its `-f` option.

> * Perl is widely available, and its [`Time::Piece` module][time-piece] is can be used to parse timestamps.

Reviewer: ```suggestion
* Perl is widely available, and its [`Time::Piece` module][time-piece] can be used to parse timestamps.
```

# placeholder

## How to solve with bash

`bash` does not have any builtin ways to **parse** datetime strings.

> This exercise requires you to call out to some external utility.

Reviewer: ```suggestion
This exercise requires you to call out to an external utility.
```

* The GNU `date` command's `-d` option can parse [a wide variety of formats][gnu-date].

> * If you're on a Mac, the [BSD `date` command][bsd-date] and its `-f` option.

Reviewer: ```suggestion
* For Mac users, see the [BSD `date` command][bsd-date] and its `-f` option.
```

> * Perl is widely available, and it's [`Time::Piece` module][time-piece] is suitable.

Reviewer: ```suggestion
* Perl is widely available, and its [`Time::Piece` module][time-piece] is can be used to parse timestamps.
```

> bash can **format** dates though.
> Given an integer time value (an epoch time value), [the `printf` `%()T` formatter][bash-printf] can be given a "strftime"-style format string.
> An example:

Reviewer: ```suggestion
bash can **format** dates without any external tools.
Given an integer time value (seconds since the epoch), [the `printf` `%()T` formatter][bash-printf] can apply "strftime"-style formatting.
```

```bash
epoch=1234567890

printf '%(%Y-%m-%d %H:%M:%S)T\n' "$epoch"
# ==> 2009-02-13 18:31:30

> # Even in different timezones

Reviewer: ```suggestion
# bash uses the `TZ` environment variable for the timezone.
```
